node/src/main/scala/coop/rchain/node/api/WebApi.scala

```diff
@@ -308,6 +305,30 @@ object WebApi {
       ExprMap(fields.toMap).some
     } else none
 
+  private def parProtoFromExpr(exp: RhoExpr): Par = {
+    def exprToPar(exp: Expr): Par = Par(exprs = Seq(exp))
+
+    exp match {
+      // Nested expressions (Par, Tuple, List and Set are converted to JSON list)
+      case ExprPar(data)   => exprToPar(Expr().withEListBody(EList(data.map(parProtoFromExpr))))
+      case ExprTuple(data) => exprToPar(Expr().withETupleBody(ETuple(data.map(parProtoFromExpr))))
+      case ExprList(data)  => exprToPar(Expr().withEListBody(EList(data.map(parProtoFromExpr))))
+      case ExprSet(data)   => exprToPar(Expr().withESetBody(ParSet(data.map(parProtoFromExpr))))
+      case ExprMap(data) =>
+        exprToPar(Expr().withEMapBody(ParMap(data.map {
+          case (k, v) => (parProtoFromExpr(ExprString(k)), parProtoFromExpr(v))
+        }.toList)))
+      // Terminal expressions (here is the data)
+      case ExprBool(data)   => exprToPar(Expr().withGBool(data))
+      case ExprInt(data)    => exprToPar(Expr().withGInt(data))
+      case ExprString(data) => exprToPar(Expr().withGString(data))
+      case ExprUri(data)    => exprToPar(Expr().withGUri(data))
+      // Binary data is encoded as base16 string
```

Comment:
Maybe it would be more clear to say that data is decoded from base 16 string?

Reply:
Yes, make sense. I [changed](https://github.com/rchain/rchain/pull/3718/commits/1dc98e2df15208b1428b042939442f6e8d1e5e5d) comment.